genomepy/annotation.py

```diff
@@ -0,0 +1,395 @@
+import gzip
+import os
+import re
+import shutil
+import subprocess as sp
+import sys
+from tempfile import mkdtemp
+from typing import Optional, Union
+
+from genomepy.utils import (
+    get_genomes_dir,
+    gzip_and_name,
+    gunzip_and_name,
+    glob_ext_files,
+    update_readme,
+    rm_rf,
+)
+
+
+class Annotation:
+    """
+    Gene annotation related functions
+
+    Parameters
+    ----------
+    name : str
+        Genome name
+
+    genomes_dir : str
+        Genomes installation directory
+
+    Returns
+    -------
+    Annotation object
+    """
+
+    def __init__(self, name: str, genomes_dir: str = None):
+        self.name = name
+        self.genome_dir = os.path.join(get_genomes_dir(genomes_dir), name)
+        self.readme_file = os.path.join(self.genome_dir, "README.txt")
+        self.annotation_gtf_file = self._get_genome_file("annotation.gtf")
+        self.annotation_bed_file = self._get_genome_file("annotation.bed")
+
+        # variables that may be None
+        self.genome_file = self._get_genome_file("fa", check_exists=False)
+        self.index_file = self.genome_file + ".fai" if self.genome_file else None
+        self.sizes_file = self.genome_file + ".sizes" if self.genome_file else None
+
+        # variables to store if called
+        self.genome_contigs = []
+        self.annotation_contigs = []
+
+    def _get_genome_file(self, ext: str, check_exists: Optional[bool] = True):
+        """
+        Returns the filepath to a single (gzipped) file in the genome_dir with matching ext.
+        """
+        ext_files = glob_ext_files(self.genome_dir, ext)
+        fname = [f for f in ext_files if f"{self.name}.{ext}" in f]
+        if len(fname) == 1:
+            filepath = os.path.join(self.genome_dir, fname[0])
+            return filepath
+
+        # error handling
+        if len(fname) > 1:
+            raise IndexError(
+                f"Too many ({len(fname)}) files match '{self.name}.{ext}(.gz)' "
+                f"in directory {self.genome_dir}"
+            )
+        if check_exists:
+            raise FileNotFoundError(
+                f"could not find '{self.name}.{ext}(.gz)' in directory {self.genome_dir}"
+            )
+        return None
+
+    @property
+    def genome_contigs(self):
+        return self.__genome_contigs
```

Comment:
any reason to use double underscore (`_`) here? I don't know what is more correct, but I believe accessing the double underscored attrs is much more difficult outside of the class than the single one